app/src/main/kotlin/com/hippo/ehviewer/ui/screen/DownloadsScreen.kt

```diff
@@ -309,50 +357,65 @@ fun DownloadsScreen(navigator: DestinationsNavigator) = composing(navigator) {
+                                                },
+                                                onDragStopped = {
+                                                    if (fromIndex != -1) {
+                                                        if (fromIndex != index) {
+                                                            val range = if (fromIndex < index) fromIndex..index else index..fromIndex
+                                                            val toUpdate = labelList.slice(range)
+                                                            toUpdate.zip(range).forEach { it.first.position = it.second }
+                                                            launchIO { EhDB.updateDownloadLabel(toUpdate) }
+                                                        }
+                                                        fromIndex = -1
                                                     }
-                                                }
-                                            },
-                                        ) {
-                                            Icon(imageVector = Icons.Default.Edit, contentDescription = null)
+                                                },
+                                            )
                                         }
-                                        DragHandle(
-                                            onDragStarted = {
-                                                fromIndex = index
-                                            },
-                                            onDragStopped = {
-                                                if (fromIndex != -1) {
-                                                    if (fromIndex != index) {
-                                                        val range = if (fromIndex < index) fromIndex..index else index..fromIndex
-                                                        val toUpdate = labelList.slice(range)
-                                                        toUpdate.zip(range).forEach { it.first.position = it.second }
-                                                        launchIO { EhDB.updateDownloadLabel(toUpdate) }
-                                                    }
-                                                    fromIndex = -1
-                                                }
-                                            },
-                                        )
                                     }
                                 },
                             )
                         }
+                        when (filterMode) {
+                            DownloadsFilterMode.ARTIST -> content()
+                            DownloadsFilterMode.CUSTOM -> SwipeToDismissBox2(
+                                state = dismissState!!,
+                                backgroundContent = {},
+                                content = { content() }
```

Comment:
Avoid creating extra lambda wrapper here

Reply:
```kotlin
val content: @Composable RowScope.() -> Unit = {..}
when (filterMode) {
    DownloadsFilterMode.ARTIST -> content(/* A RowScope cannot be provided */)
    DownloadsFilterMode.CUSTOM -> SwipeToDismissBox2(
        state = dismissState!!,
        backgroundContent = {},
        content = content
    )
}
```
I know it's probably stupid, but I can't think of anything idea, can i get some help? thanks

Reply:
Oops, I didn't look into it, just in case it can be passed directly.
That's fine, let's keep it as is